### ISSUE TITLE
Ludwig reports that validation loss improved while it shows it to increase - confusing

### ISSUE THREAD

**Issue by jmizgajski** [uber/ludwig#734]
**Describe the bug**
@w4nderlust 

I am training a tagger [like in here](https://github.com/uber/ludwig/issues/729). I also specify class weights to force it to be more aggressive in assigning the class `X`. (to be honest I am not sure, if it takes effect)

```
(...)   
   loss:
      type: softmax_cross_entropy
      class_weights:
        X: 1.
        O: 0.02
```

What confuses me is that in the reports ludwig prints out I see the validation loss increase but ludwig reports that the loss improved (which I would assume would mean for it to decrese), saves the model and continues training. Does ludwig make the comparison on some different value than it prints out?

```
#  loss is in first numeric column

Epoch   8
────────────────┼──────────────┼─────────────────┤
│ vali              │ 0.0885 │     0.3998 │           0.9671 │          0.9530 │       1.0925 │          0.0347 │
├───────────────────┼────────┼────────────┼──────────────────┼
Validation loss on combined improved, model saved

Epoch   9
─────────────────┼──────────────┼─────────────────┤
│ vali              │ 0.0990 │     0.4143 │           0.9654 │          0.9542 │       1.1041 │          0.0363 │
├───────────────────┼────────┼────────────┼──────────────────┼ 
 Validation loss on combined improved, model saved

```

**To Reproduce**

I cannot provide the data I train on, but I observed it frequently, so maybe it can be seen on some other dataset you test on. I may try to provide synthetic data in the future, but maybe the problem can be diagnosed without it just looking at the code.

**Expected behavior**
Save the model and report improvement only when the validation loss truly decreses and it can be noticed in the console report.

**Screenshots**

**Environment (please complete the following information):**
Centos7
- Python version
3.6.8
- Ludwig version
current master

**Additional context**

**Comment by w4nderlust** [comment]
A bunch of things here:
1) the weighting on the classes should work, I can doublecheck to be sure, will look into it. In the mean time you can try using "sampled_softmax_cross_entropy" as loss where it should also make a difference.
2) the improvement is calculated on a "combined" output feature by default. The loss of "combined" is the sum of the losses of the individual output features plus regularization. Becuase I believe you have only one class what i suspect is happening is that you also added regularization or other parts of the loss that make the value of the loss of "combined" different from the value of the loss of your output feature. I can't be sure without doublechecking with a reproducible example thoug. One thing you can do to check for this hypothesys is to set the training validation_field to "tags" (which I believe is the name of your output feature) and there should be no misalignment. i see how this could be confusing as it is probably not clearly explained in the docs, I will think about revising the description accordingly and also probably print the value of combined (it is printed when there's more than one feature and hidden when there's only one at the moment, but if you look at the output training statistics you'll find it there).
3) If you can reproduce the issue with synthetic data that would be of great help for me to go and fix the probem if there is one.

**Comment by jmizgajski** [comment]
I did add regularization in my settings so this may be the case. I think printing the combined loss even if one feature is present would probably do it. Thank you.

**Comment by w4nderlust** [comment]
Good, so after looking at the combined loss in the training_statistics.json file, can you confirm that the early stopping was performed at the right time?
Just to be sure there is not any logic issue, it's ont a reporting issue.

**Comment by w4nderlust** [comment]
We fixed thsi and changed the logic, improved prints and now it should work fine. Let me know if you still have problems.
